File: algorithms/Two Sum IV - Input is a BST/Two Sum IV - Input is a BST.java
class Solution {
    Set<Integer> set = new HashSet<>();
    public boolean findTarget(TreeNode root, int k) {
        if(root == null){
            return false;
        }
        if(set.contains(k-root.val)){
            return true;
        }
        set.add(root.val);
        return findTarget(root.left,k) || findTarget(root.right,k);
    }
}

File: scripts/algorithms/T/Two Sum IV - Input is a BST/Two Sum IV - Input is a BST.js
// Runtime: 97 ms (Top 93.88%) | Memory: 51.4 MB (Top 81.78%)
var findTarget = function(root, k) {

    const set = new Set();

    const search = (root, k) => {
        if (!root) return false;
        if (set.has(k - root.val)) return true;
        set.add(root.val);
        return search(root.left, k) || search(root.right, k);
    }

    return search(root,k);
};